service/matching/version_sets.go

```diff
@@ -244,86 +241,77 @@ func updateImpl(timestamp hlc.Clock, existingData *persistencespb.VersioningData
 				if addNew.GetMakeSetDefault() && targetSetIdx != numExistingSets-1 {
 					return nil, serviceerror.NewInvalidArgument(fmt.Sprintf("version %s already exists and is not default for queue", targetedVersion))
 				}
-				if versionInSetIdx != len(existingData.GetVersionSets()[targetSetIdx].BuildIds)-1 {
+				if versionInSetIdx != len(data.GetVersionSets()[targetSetIdx].BuildIds)-1 {
 					return nil, serviceerror.NewInvalidArgument(fmt.Sprintf("version %s already exists and is not default in set", targetedVersion))
 				}
 				// Make the operation idempotent
-				return existingData, nil
+				return data, nil
 			}
 			return nil, serviceerror.NewInvalidArgument(fmt.Sprintf("%s requested to be made compatible with %s but both versions exist and are incompatible", targetedVersion, compatVer))
 		}
 
 		// First duplicate the build IDs to avoid mutation
```

Comment:
This comment is wrong now that it's not being cloned

Reply:
thanks